.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 New Relic Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+#
+# This is the workflow to do a PR build.
+#
+
+name: PR_CI
+
+#
+# Controls when the action will run. 
+#
+on:
+  #
+  # Triggers the workflow on push or pull request events but only for the main branch
+  #
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  agent_pr:
+    env:
+      PHP_VER: ${{ matrix.php_ver }}
+      ARCH: ${{ matrix.arch }}
+      BUILD_TYPE: pr
+      OS: ${{ matrix.os }}
+      image_name: ${{ matrix.os }}:${{ matrix.php_ver }}${{ matrix.arch }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [linux]
+        php_ver: ['8.0', '8.0-zts']
+        arch: ['x64', 'x86']
+    steps:
+      - name: Checkout Repo 
+        uses: actions/checkout@v2
+      - name: Build Custom Docker Image
+        run: docker build --build-arg ARCH=$ARCH --build-arg PHP_VER=$PHP_VER --build-arg BUILD_TYPE=$BUILD_TYPE -f ./.github/docker/${OS}/${ARCH}/Dockerfile -t $image_name .
+      - name: Build and Test
+        run: docker run --name runtest --workdir /github/workspace --rm -e GITHUB_WORKSPACE -e GITHUB_ENV -v "${GITHUB_WORKSPACE}":"/github/workspace" $image_name
```

Comment:
Nicely done! I struggled to figure out how to reduce the number of `Dockerfile`s we had to maintain (and the resulting verbosity of the matrix to use them). This does that beautifully!